src/pg/sql/42_observatory_exploration.sql

```diff
@@ -372,16 +372,28 @@ BEGIN
              geom_type::TEXT,
              geom_extra::JSONB,
              geom_tags::JSONB,
-      $1 = ANY(numers) valid_numer,
-      $2 = ANY(denoms) valid_denom,
-      $3 = ANY(timespans) valid_timespan
-      FROM observatory.obs_meta_geom
+             $1 = ANY(numers) valid_numer,
+             $2 = ANY(denoms) valid_denom,
+             CASE WHEN $3 IS NOT NULL AND $3 != '' THEN
+              -- Here we are looking for geometries with: a) geometry timespan or b) numerators linked to that geometries that fit in the
+              -- timespan passed. For example it look for geometries with timespan '2015 - 2015' or numerators linked to that geometry that has
+              -- '2015 - 2015' as one of the valid timespans.
+              -- If we pass a numerator_id, we filter by that numerator
+              CASE WHEN $1 IS NOT NULL AND $1 != '' THEN
+                EXISTS (SELECT 1 FROM observatory.obs_meta_geom_numer_timespan onu WHERE o.geom_id = onu.geom_id AND onu.numer_id = $1 AND ($3 = ANY(onu.timespans) OR $3 IN (select(unnest(o.timespans)))))
+              ELSE
+                EXISTS (SELECT 1 FROM observatory.obs_meta_geom_numer_timespan onu WHERE o.geom_id = onu.geom_id AND ($3 = ANY(onu.timespans) OR $3 IN (select(unnest(o.timespans)))))
```

Comment:
Use of IN instead of ANY makes the function change from Bitmap scan to Index Scan (read it [here](https://coderwall.com/p/jmtskw/use-in-instead-of-any-in-postgresql))

I'm pretty sure we are able to optimize it more